app/src/main/java/com/infomaniak/drive/ui/fileShared/FileSharedListFragment.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Infomaniak kDrive - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.drive.ui.fileShared
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import androidx.core.view.isGone
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.infomaniak.drive.R
+import com.infomaniak.drive.data.models.File
+import com.infomaniak.drive.ui.fileList.FileListFragment
+import com.infomaniak.drive.ui.fileShared.FileSharedViewModel.Companion.ROOT_SHARED_FILE_ID
+import com.infomaniak.drive.utils.FilePresenter.displayFile
+import com.infomaniak.drive.utils.FilePresenter.openBookmark
+import com.infomaniak.drive.utils.FilePresenter.openFolder
+
+class FileSharedListFragment : FileListFragment() {
+
+    private val fileShareViewModel: FileSharedViewModel by activityViewModels()
```

Comment:
I can't, because if so its lifecycle will be scoped to the fragment, but I want it to be scoped to the activity
If it's scoped to the fragment, each time we navigate to it, it will have a new instance and I don't want that